tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -12230,13 +12265,16 @@ bool SpirvEmitter::emitEntryFunctionWrapper(const FunctionDecl *decl,
 
   // Create temporary variables for holding function call arguments
   llvm::SmallVector<SpirvInstruction *, 4> params;
-  for (const auto *param : decl->params()) {
+  for (auto *param : decl->params()) {
     const auto paramType = param->getType();
     std::string tempVarName = "param.var." + param->getNameAsString();
     auto *tempVar =
         spvBuilder.addFnVar(paramType, param->getLocation(), tempVarName,
                             param->hasAttr<HLSLPreciseAttr>());
 
+    if (paramType != tempVar->getAstResultType())
+        param->setType(tempVar->getAstResultType());
```

Comment:
In spvBuilder.addFnVar call here:
```
        spvBuilder.addFnVar(paramType, param->getLocation(), tempVarName,
                            param->hasAttr<HLSLPreciseAttr>());
```
I add code for pervertexExt decorated input (COLOR).
This input will be changed to a three-element array according to extension spec (when builtin function GetAttributeAtVertex has been detected before).
```
if (astContext->validateBaryCoordInputLoc(loc)) {
      if (!valueType->isArrayType())
      {
        QualType qtype = astContext->getConstantArrayType(
            valueType, llvm::APInt(32, 3), clang::ArrayType::Normal, 0);
        var = new (context) SpirvVariable(
              qtype, loc, spv::StorageClass::Function, isPrecise, init);
      }
      else
        var = new (context) SpirvVariable(
            valueType, loc, spv::StorageClass::Function, isPrecise, init);
    }
```
Here is to change param type as same as generated spv variable.
Otherwise, we will get a type mismatched error later (OpLoad for function parameter).